File: schedule-web/src/types/goal.ts
export interface Goal {
  id: number;
  userId: number;
  rawInput: string;
  parsedTarget: string;
  parsedDeadline: string;
  parsedItems: string;
  status: 'PARSING' | 'CLARIFYING' | 'PARSED' | 'CONFIRMED';
  clarifyRound: number;
  partialResult: string;
  conversation: string;
  createdAt: string;
  updatedAt: string;
}

export interface GoalParseVO {
  goalId: number;
  status: string;
  parsedTarget: string;
  parsedDeadline: string;
  parsedItems: string;
  confidence: number;
  needsClarification: boolean;
  questions: string[];
  currentRound: number;
}

export interface GoalParseDTO {
  rawInput: string;
}

export interface GoalReplyDTO {
  reply: string;
}

File: schedule-server/src/main/java/com/aspire/schedule/repository/entity/Goal.java
package com.aspire.schedule.repository.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("goal")
public class Goal {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String rawInput;
    private String parsedTarget;
    private LocalDateTime parsedDeadline;
    private String parsedItems;
    private String status;
    private Integer clarifyRound;
    private Integer scheduleRound;
    private String partialResult;
    private String conversation;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
